### DISCUSSION TITLE
It's not possible to update multiple items with different data anymore?

### DISCUSSION THREAD

**Discussion by DaanMoura** [directus/directus#5530] [Archive]
I used to update multiple items at the same time with different data, just sending an array with id and the data. But now, I am getting an error. I checked the docs and seems that I have to send the keys on the "keys" field, and the data on "data". 

https://docs.directus.io/reference/api/items/#update-multiple-items

That's ok when I need to update with the same data, but not with different data. There is a way to do what I want, or I have to iterate my array of items and update one by one?

(I'm using 9.0.0-rc.61)

**Comment by rijkvanzanten** [comment]
> There is a way to do what I want, or I have to iterate my array of items and update one by one?

No, that's not supported so you'll have to iterate over and do it that way 👍🏻

**Reply by northern-mindset** [reply]
What is the rationale behind this decision?

In contrast, creating multiple items in one request is supported: https://docs.directus.io/reference/items.html#example-4

**Reply by st11x** [reply]
I was able to update multiple items by passing an array of items now (on 10.9.3). But for relations, you have to pass all the data for that item, you cannot append.
